src/lib/generate-files.ts

```diff
@@ -67,14 +84,24 @@ export async function generateFiles(
       target,
       remaps,
       filePath,
-      substitutions
+      substitutions,
     );
 
+    console.log(filePath, computedPath);
+
     const newFileContent = render(filePath, substitutions);
-    const formattedContent = await prettier.format(newFileContent, { parser: options.parser || 'typescript' });
+
+    // allow sending a function to determine the parser based on the file extension
+    const prettierParser =
+      typeof options.parser === "function"
+        ? options.parser(computedPath)
+        : options.parser || "typescript";
+    const formattedContent = await prettier.format(newFileContent, {
+      parser: prettierParser,
+    });
```

Comment:
And here.. is where I'm switching between the function or just the static string